### PR TITLE
Refactor: Recycler View with List Adapter

### PR DESCRIPTION
**Pull request by sagar-viradiya**
- Changed the PlantAdapter and GardenPlantingAdapter to new list adapter which offers animation and diff util to improve performance.

- New DiffUtil implementation for PlantAdapter and GardenPlantAdapter having item amd content comparison.

### CODE REVIEW

File: app/src/main/java/com/google/samples/apps/sunflower/adapters/GardenPlantDiffCallback.kt [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ */
+
+package com.google.samples.apps.sunflower.adapters
+
+import android.support.v7.util.DiffUtil
+import com.google.samples.apps.sunflower.data.PlantAndGardenPlantings
+
+class GardenPlantDiffCallback : DiffUtil.ItemCallback<PlantAndGardenPlantings>() {
+
+    override fun areItemsTheSame(oldItem: PlantAndGardenPlantings?,
```
Comment (tiembo) [inline app/src/main/java/com/google/samples/apps/sunflower/adapters/GardenPlantDiffCallback.kt]: nit: formatting -

    override fun areItemsTheSame(
            oldItem: PlantAndGardenPlantings?,
            newItem: PlantAndGardenPlantings?
    ): Boolean {

File: app/src/main/java/com/google/samples/apps/sunflower/adapters/GardenPlantDiffCallback.kt [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+                                 newItem: PlantAndGardenPlantings?): Boolean {
+        return oldItem?.plant?.plantId == newItem?.plant?.plantId
+    }
+
+    override fun areContentsTheSame(oldItem: PlantAndGardenPlantings?,
```
Comment (tiembo) [inline app/src/main/java/com/google/samples/apps/sunflower/adapters/GardenPlantDiffCallback.kt]: nit: formatting -

    override fun areContentsTheSame(
            oldItem: PlantAndGardenPlantings?,
            newItem: PlantAndGardenPlantings?
    ): Boolean {

**Comment by sagar-viradiya** [comment]
Done with changes

**Comment by tiembo** [comment]
Thanks!

**Pull request merged**